docs/PythonAPI.md

```diff
@@ -0,0 +1,867 @@
+
+- `line_interval_us`: The time to scan one line in microseconds in a rolling shutter camera camera.
+
+- `binning`: How many adjacent pixels in each direction to combine by averaging. If binning is set to 1, no pixels will be combined.
+
+- `pixel_type`: An instance of the `SampleType` class which specifies the numerical data type.
+
+- `readout_direction`: An instance of the `Direction` class which specifies whether the data is readout forwards or backwards.
+
+- `offset`: A tuple of two integers representing the dark counts of the camera.
+
+- `shape`: A tuple of two integers representing the shape of the camera detector in pixels.
+
+- `input_triggers`: An instance of the `InputTriggers` class.
+
+- `output_triggers`: An instance of the `OutputTriggers` class.
+
+- The `dict` method create a dictionary of a `CameraProperties` object's attributes.
+
+## Class `ChunkingProperties`
+The `ChunkingProperties` class represents properties related to data chunking for storage in a Zarr container.
+
+```python
+class ChunkingProperties:
+    max_bytes_per_chunk: int
+    tile: TileShape
+
+    def dict(self) -> Dict[str, Any]: ...
+    """Returns a dictionary of the ChunkingProperties attributes."""
+```
+
+- `max_bytes_per_chunk`: The maximum number of bytes per data chunk.
+
+- `tile`: An instance of the `TileShape` class representing the shape of the data chunk tile.
+
+- The `dict` method creates a dictionary of a `ChunkingProperties` object's attributes.
+
+## Class `DeviceIdentifier`
+The `DeviceIdentifier` class represents an identifier for a device, such as a camera or video source.
+
+```python
+class DeviceIdentifier:
+    id: Tuple[int, int]
+    kind: DeviceKind
+    name: str
+
+    def __init__(self, *args: None, **kwargs: Any) -> None: ...
+    """Initializes a DeviceIdentifier object with optional arguments."""
+
+    def dict(self) -> Dict[str, Any]: ...
+    """Returns a dictionary of the DeviceIdentifier attributes."""
+
+    @staticmethod
+    def none() -> DeviceIdentifier: ...
+    """Returns a "None" type DeviceIdentifier. Useful when a DeviceIdentifier is not needed."""
+
+    def __eq__(self, other: object) -> bool:
+        """Checks if two DeviceIdentifier objects are equal."""
+        
+    def __ge__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is greater than or equal to another."""
+        
+    def __gt__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is greater than another."""
+        
+    def __le__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is less than or equal to another."""
+        
+    def __lt__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is less than another."""
+        
+    def __ne__(self, other: object) -> bool:
+        """Checks if two DeviceIdentifier objects are not equal."""
+```
+
+- `id`: A tuple containing two integers that serve as the device ID.
```

Comment:
What do the integers represent?

Reply:
I am not sure. All I could find is that they are u8 integers. Here is where I was looking: https://github.com/acquire-project/acquire-python/blob/cb2dd9bdec0f483c3391b54ef8996c570614379d/src/device.rs#L92